src/parser.js

```diff
@@ -1693,14 +1713,17 @@ export class Parser extends Tokenizer {
         return this.markLocation(new Shift.ShorthandProperty(new Shift.Identifier(key.value)), startTokenIndex);
       }
     }
-    if (this.eof() || token.type.klass == TokenClass.Punctuator) {
-      throw this.createUnexpected(token);
-    } else {
-      let key = this.parseObjectPropertyKey();
-      this.expect(TokenType.COLON);
+
+    let key = this.parseObjectPropertyKey();
+    if (this.eat(TokenType.COLON)) {
+      // PropertyName[?Yield] : AssignmentExpression[In,?Yield]
       let value = this.parseAssignmentExpression();
       return this.markLocation(new Shift.DataProperty(key, value), startTokenIndex);
     }
+    this.match(TokenType.LPAREN);
+    let parmInfo = this.parseParams(null);
```

Comment:
Do you think it would be better to extract this and the next few lines to a `parseMethod` function so that we can remove the duplication from a few lines above?